ipu-plugin/pkg/ipuplugin/lifecycleservice.go

```diff
@@ -194,30 +195,90 @@ func getCommPf(mode string, linkList []netlink.Link) (netlink.Link, error) {
 }
 
 /*
-It can take time for network-manager's state for each interface, to become
-activated, when IP address is set, which can cause the IP address to not stick.
-Removing the host-acc comm channel interface out of network-manager's mgmt.
+Upon host reboot, it seems to take time for network-manager's state for each interface,
+to stabilize and become activated. This function does a few retries,
+when it tries to set IP.
 TODO: Currently we only support nmcli/NetworkManager daemon combination(RHEL),
 this api can be extended for other distros that use different CLI/systemd-networkd.
 */
-func (e *ExecutableHandlerImpl) nmcliSetup(link netlink.Link) error {
+func (e *ExecutableHandlerImpl) nmcliSetIPAddress(link netlink.Link, ipAddr string) error {
 	intfName := link.Attrs().Name
-	output, err := utils.ExecuteScript(`nmcli general status`)
-	if err == nil {
-		output, err = utils.ExecuteScript(`nmcli device set ` + intfName + ` managed no`)
+	ipWithMask := ipAddr + "/24"
+	var runCmd string
+	var ipAddrSet bool
+	var err error
+	var output string
+	maxRetries := 5
+	retryInterval := 10 * time.Second
+
+	for cnt := 0; cnt < maxRetries; cnt++ {
+		_, err = utils.ExecuteScript(`nmcli general status`)
 		if err != nil {
-			log.Errorf("nmcli err->%v, output->%v\n", err, output)
-			return fmt.Errorf("nmcli err->%v, output->%v\n", err, output)
+			log.Infof("network manager not running, nmcli, err->%v\n", err)
+			goto retry
 		}
-	} else {
-		log.Infof("network manager not running, skipping nmcli, err->%v\n", err)
-	}
-	output, err = utils.ExecuteScript(`ip link set ` + intfName + ` up`)
-	if err != nil {
-		log.Errorf("ip link set err->%v, output->%v\n", err, output)
-		return fmt.Errorf("ip link set  err->%v, output->%v\n", err, output)
+		runCmd = fmt.Sprintf(`nmcli conn show | grep -w "%s"`, intfName)
```

Comment:
nmcli conn show should also specify intfName...otherwise...we end up making invalid assumption...that interface already exists....when actually....that interface name just happens...to show up in a different coloumn...as part of the output.